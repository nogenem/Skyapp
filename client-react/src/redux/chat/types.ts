import { TMessageType } from '~/constants/message_types';
import { TUserStatus } from '~/constants/user_status';

interface IOtherUser {
  _id: string;
  nickname: string;
  thoughts: string;
  status: TUserStatus;
  online: boolean;
  channel_id?: string;
}

interface IOtherUsers {
  [_id: string]: IOtherUser;
}

interface IMember {
  user_id: string;
  is_adm: boolean;
  last_seen: Date;
}

interface IChannel {
  _id: string;
  name: string;
  is_group: boolean;
  members: IMember[];
  other_member_idx?: number;
  unread_msgs: number;
  lastMessage?: IMessage;
}

interface IChannels {
  [_id: string]: IChannel;
}

interface IAttachment {
  originalName: string;
  size: number;
  path: string;
  mimeType: string;
  imageDimensions?: { width: number; height: number };
}

interface IMessage {
  _id: string;
  channel_id: string;
  from_id?: string;
  body: string | IAttachment;
  type: TMessageType;
  createdAt: Date;
  updatedAt: Date;
}

interface IInitialData {
  users: IOtherUsers;
  channels: IChannels;
}

interface IActiveChannelInfo {
  _id: string;
  messages: IMessage[];
  totalMessages: number;
  queue: IMessage[];
}

interface INewGroupCredentials {
  name: string;
  members: string[];
  admins: string[];
}

interface IUpdateGroupCredentials {
  channel_id: string;
  name: string;
  members: string[];
  admins: string[];
}

interface ILeaveGroupCredentials {
  channel_id: string;
}

interface IFetchMessagesCredentials {
  channel_id: string;
  offset: number;
  limit?: number;
  sort?: string;
}

interface ISendMessageCredentials {
  channel_id: string;
  body: string;
}

type TChatState = {
  users: IOtherUsers;
  channels: IChannels;
  activeChannelInfo?: IActiveChannelInfo;
};

enum EChatActions {
  SET_INITIAL_DATA = '@chat/SET_INITIAL_DATA',
  SET_USER_ONLINE = '@chat/SET_USER_ONLINE',
  ADD_OR_UPDATE_CHANNEL = '@chat/ADD_OR_UPDATE_CHANNEL',
  SET_ACTIVE_CHANNEL = '@chat/SET_ACTIVE_CHANNEL',
  REMOVE_CHANNEL = '@chat/REMOVE_CHANNEL',
  ADD_NEW_USER = '@chat/ADD_NEW_USER',
  ADD_MESSAGES = '@chat/ADD_MESSAGES',
  SET_LATEST_MESSAGE = '@chat/SET_LATEST_MESSAGE',
  ADD_TO_MESSAGES_QUEUE = '@chat/ADD_TO_MESSAGES_QUEUE',
  REMOVE_FROM_MESSAGES_QUEUE = '@chat/REMOVE_FROM_MESSAGES_QUEUE',
  SET_LAST_SEEN = '@chat/SET_LAST_SEEN',
  SET_USER_STATUS = '@chat/SET_USER_STATUS',
  SET_USER_THOUGHTS = '@chat/SET_USER_THOUGHTS',
}

interface IChatActionType<T, P> {
  type: T;
  payload: P;
}

type TChatAction =
  | IChatActionType<typeof EChatActions.SET_INITIAL_DATA, IInitialData>
  | IChatActionType<
      typeof EChatActions.SET_USER_ONLINE,
      { _id: string; value: boolean }
    >
  | IChatActionType<typeof EChatActions.ADD_OR_UPDATE_CHANNEL, IChannel>
  | IChatActionType<
      typeof EChatActions.SET_ACTIVE_CHANNEL,
      { _id: string | undefined }
    >
  | IChatActionType<typeof EChatActions.REMOVE_CHANNEL, { channelId: string }>
  | IChatActionType<typeof EChatActions.ADD_NEW_USER, IOtherUser>
  | IChatActionType<
      typeof EChatActions.ADD_MESSAGES,
      { messages: IMessage[]; totalMessages: number; atTop: boolean }
    >
  | IChatActionType<typeof EChatActions.SET_LATEST_MESSAGE, IMessage>
  | IChatActionType<typeof EChatActions.ADD_TO_MESSAGES_QUEUE, IMessage>
  | IChatActionType<typeof EChatActions.REMOVE_FROM_MESSAGES_QUEUE, IMessage>
  | IChatActionType<
      typeof EChatActions.SET_LAST_SEEN,
      {
        channel_id: string;
        user_id: string;
        last_seen: string;
      }
    >
  | IChatActionType<
      typeof EChatActions.SET_USER_STATUS,
      {
        user_id: string;
        newStatus: TUserStatus;
      }
    >
  | IChatActionType<
      typeof EChatActions.SET_USER_THOUGHTS,
      {
        user_id: string;
        newThoughts: string;
      }
    >;

export type {
  TChatState,
  TChatAction,
  IInitialData,
  IOtherUsers,
  IOtherUser,
  IChannel,
  IChannels,
  IMember,
  IMessage,
  IAttachment,
  INewGroupCredentials,
  IUpdateGroupCredentials,
  ILeaveGroupCredentials,
  IFetchMessagesCredentials,
  ISendMessageCredentials,
};
export { EChatActions };
