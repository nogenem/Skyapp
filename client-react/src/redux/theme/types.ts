type TThemeMode = 'dark' | 'light';

type TThemeState = {
  mode?: TThemeMode;
};

enum EThemeActions {
  MODE_SWITCHED = '@theme/MODE_SWITCHED',
}

interface IThemeActionType<T, P> {
  type: T;
  payload: P;
}

type TThemeAction = IThemeActionType<
  typeof EThemeActions.MODE_SWITCHED,
  TThemeMode
>;

export type { TThemeMode, TThemeState, TThemeAction };
export { EThemeActions };
