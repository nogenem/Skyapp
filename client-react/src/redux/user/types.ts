interface User {
  _id: string;
  token?: string;
}

type UserState = User;

enum EUserActions {
  SIGNED_IN = '@user/SIGNED_IN',
  SIGNED_OUT = '@user/SIGNED_OUT',
}

interface UserActionType<T, P> {
  type: T;
  payload: P;
}

type UserAction =
  | UserActionType<typeof EUserActions.SIGNED_IN, User>
  | UserActionType<typeof EUserActions.SIGNED_OUT, null>;

export type { User, UserState, UserAction };
export { EUserActions };
