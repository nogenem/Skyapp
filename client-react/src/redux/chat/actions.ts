import { Dispatch } from 'redux';

import * as SOCKET_EVENTS from '~/constants/socket_events';
import type { IUser } from '~/redux/user/types';
import api from '~/services/api';
import io from '~/services/io';

import type {
  IChannel,
  IInitialData,
  INewGroupCredentials,
  IOtherUser,
  IUpdateGroupCredentials,
} from './types';
import { EChatActions } from './types';

const setInitialData = (data: IInitialData) => ({
  type: EChatActions.SET_INITIAL_DATA,
  payload: data,
});

const setUserOnline = (_id: string, value: boolean) => ({
  type: EChatActions.SET_USER_ONLINE,
  payload: {
    _id,
    value,
  },
});

const addNewChannel = (channel: IChannel) => ({
  type: EChatActions.ADD_NEW_CHANNEL,
  payload: channel,
});

export const setActiveChannel = (channel_id: string) => ({
  type: EChatActions.SET_ACTIVE_CHANNEL,
  payload: { _id: channel_id },
});

export const connectIo = (user: IUser) => (dispatch: Dispatch) => {
  const instance = io.instance();
  instance.connect({ _id: user._id });

  instance.socket!.on(SOCKET_EVENTS.SOCKET_CONNECT, () => {
    instance.socket!.emit(
      SOCKET_EVENTS.IO_GET_INITIAL_DATA,
      (data: IInitialData) => {
        dispatch(setInitialData(data));
      },
    );

    instance.socket!.on(SOCKET_EVENTS.IO_SIGNIN, (_id: string) => {
      dispatch(setUserOnline(_id, true));
    });
    instance.socket!.on(SOCKET_EVENTS.IO_SIGNOUT, (_id: string) => {
      dispatch(setUserOnline(_id, false));
    });
    instance.socket!.on(
      SOCKET_EVENTS.IO_PRIVATE_CHANNEL_CREATED,
      (channel: IChannel) => {
        dispatch(addNewChannel(channel));
      },
    );
    instance.socket!.on(
      SOCKET_EVENTS.IO_GROUP_CHANNEL_CREATED,
      (channel: IChannel) => {
        dispatch(addNewChannel(channel));
      },
    );
  });
};

export const disconnectIo = () => (dispatch: Dispatch) => {
  io.instance().disconnect();
};

export const startChattingWith = (otherUser: IOtherUser) => (
  dispatch: Dispatch,
) =>
  api.chat.createChannelWith(otherUser).then(({ channel_id }) => {
    dispatch(setActiveChannel(channel_id));
  });

export const createGroupChannel = (credentials: INewGroupCredentials) => (
  dispatch: Dispatch,
) =>
  api.chat.createGroupChannel(credentials).then(({ channel_id }) => {
    dispatch(setActiveChannel(channel_id));
  });

export const updateGroupChannel = (credentials: IUpdateGroupCredentials) => (
  dispatch: Dispatch,
) =>
  api.chat.updateGroupChannel(credentials).then(({ channel_id }) => {
    dispatch(setActiveChannel(channel_id));
  });
