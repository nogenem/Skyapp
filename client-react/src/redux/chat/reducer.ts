import produce from 'immer';
import { createSelector } from 'reselect';

import type { IAppState } from '../store';
import { EChatActions } from './types';
import type { TChatAction, TChatState } from './types';

export const initialState: TChatState = {
  users: {},
  channels: {},
};

export default function chat(
  state = initialState,
  action: TChatAction,
): TChatState {
  return produce(state, draft => {
    switch (action.type) {
      case EChatActions.SET_INITIAL_DATA:
        draft.users = action.payload.users;
        draft.channels = action.payload.channels;
        return draft;
      case EChatActions.SET_USER_ONLINE:
        if (draft.users[action.payload._id]) {
          draft.users[action.payload._id].online = action.payload.value;
        }
        return draft;
      default:
        return draft;
    }
  });
}

// SELECTORS
export const getChat = (state: IAppState) => state.chat || initialState;
export const getUsers = createSelector(getChat, data => data.users);
export const getChannels = createSelector(getChat, data => data.channels);
