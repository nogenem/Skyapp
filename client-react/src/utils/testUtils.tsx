import React from 'react';
import { Provider } from 'react-redux';

import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import type { IAppState } from '~/redux/store';

export const getMockStore = () => {
  const middlewares = [thunk];
  return configureStore(middlewares);
};

export const getRenderWithRedux = () => {
  const emptyState: Partial<IAppState> = {};
  const mockStore = getMockStore();

  return (
    ui: React.ReactNode,
    initialState: Partial<IAppState> = emptyState,
  ) => {
    const store = mockStore(initialState);
    return {
      ...render(<Provider store={store}>{ui}</Provider>),
    };
  };
};
