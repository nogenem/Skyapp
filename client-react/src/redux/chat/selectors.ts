import { createSelector } from 'reselect';

import type { IAppState } from '../store';
import { selectUserId } from '../user/selectors';
import { initialState } from './reducer';
import { IOtherUser } from './types';
import type { IChannel } from './types';

export const selectChat = (state: IAppState) => state.chat || initialState;

export const selectChatUsers = createSelector(selectChat, data => data.users);

export const selectChatChannels = createSelector(
  selectChat,
  data => data.channels,
);

export const selectChannelFromProps = (
  state: IAppState,
  { channel }: { channel: IChannel },
) => channel || {};

export const selectChatUsersList = createSelector(selectChat, data => {
  const users = Object.values(data.users);
  users.sort((a, b) => a.nickname.localeCompare(b.nickname));
  return users;
});

export const selectChatChannelsList = createSelector(selectChat, data => {
  const channels = Object.values(data.channels);
  channels.sort((a, b) => a.name.localeCompare(b.name));
  channels.sort((a, b) => {
    if (!!a.lastMessage && !!b.lastMessage) {
      return (
        b.lastMessage.updatedAt.getTime() - a.lastMessage.updatedAt.getTime()
      );
    } else if (!!a.lastMessage && !b.lastMessage) {
      return -1;
    } else if (!!b.lastMessage && !a.lastMessage) {
      return 1;
    } else {
      return 0;
    }
  });
  return channels;
});

export const selectChatUsersWithoutChannelList = createSelector(
  selectChat,
  data => {
    const users = Object.values(data.users).filter(user => !user.channel_id);
    users.sort((a, b) => a.nickname.localeCompare(b.nickname));
    return users;
  },
);

export const selectActiveChannelInfo = createSelector(
  selectChat,
  data => data.activeChannelInfo,
);

export const selectActiveChannelId = createSelector(
  selectChat,
  data => data.activeChannelInfo?._id,
);

export const selectActiveChannel = createSelector(
  selectChat,
  data =>
    data.channels[data.activeChannelInfo?._id || ''] as IChannel | undefined,
);

export const selectActiveChannelMessages = createSelector(
  selectChat,
  data => data.activeChannelInfo?.messages,
);

export const selectActiveChannelMessagesQueue = createSelector(
  selectChat,
  data => data.activeChannelInfo?.queue,
);

export const selectActiveChannelTotalMessages = createSelector(
  selectChat,
  data => data.activeChannelInfo?.totalMessages,
);

export const selectOtherUserFromChannel = createSelector(
  [selectChat, selectChannelFromProps],
  (data, channel) => {
    if (!channel.is_group) {
      return data.users[channel.members[channel.other_member_idx || 0].user_id];
    }
    return undefined;
  },
);

export const selectOtherUsersFromActiveChannel = createSelector(
  [selectChat, selectActiveChannel, selectUserId],
  (data, channel, loggedUserId) => {
    const users: IOtherUser[] = [];
    if (channel) {
      channel.members.forEach(member => {
        if (member.user_id !== loggedUserId) {
          users.push(data.users[member.user_id]);
        }
      });
    }
    return users;
  },
);
