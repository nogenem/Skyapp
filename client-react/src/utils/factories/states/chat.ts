import faker from 'faker/locale/en_US';

import { initialState } from '~/redux/chat/reducer';
import type {
  IChannel,
  IChannels,
  IOtherUser,
  IOtherUsers,
  TChatState,
} from '~/redux/chat/types';

import activeChannelInfoFactory from '../models/activeChannelInfo';
import channelFactory from '../models/channel';
import otherUserFactory from '../models/otherUser';

interface IOptions {
  useConstValues: boolean;
  useInitialState: boolean;
  usersLen: number;
  channelsLen: number;
}

export default (
  override?: Partial<TChatState>,
  options?: Partial<IOptions>,
): TChatState => {
  let usersLen = faker.datatype.number({ min: 0, max: 2 });
  if (options?.usersLen !== undefined && options.usersLen >= 0) {
    usersLen = options.usersLen;
  } else if (options?.useConstValues) {
    usersLen = 2;
  }

  let tmpOtherUsers: IOtherUser[] = [];
  if (!options?.useConstValues) {
    tmpOtherUsers = Array.from({ length: usersLen }, () => otherUserFactory());
  } else {
    tmpOtherUsers = Array.from({ length: usersLen }, (_, idx) =>
      otherUserFactory(
        { _id: `other-user-${idx}`, channel_id: `channel-${idx}` },
        { useConstValues: true },
      ),
    );
  }

  const users: IOtherUsers = {};
  tmpOtherUsers.forEach(user => {
    users[user._id] = user;
  });

  let channelsLen = faker.datatype.number({ min: 0, max: 2 });
  if (options?.channelsLen !== undefined && options.channelsLen >= 0) {
    channelsLen = options.channelsLen;
  } else if (options?.useConstValues) {
    channelsLen = 2;
  }

  let tmpChannels: IChannel[] = [];
  if (!options?.useConstValues) {
    tmpChannels = Array.from({ length: channelsLen }, () => channelFactory());
  } else {
    tmpChannels = Array.from({ length: channelsLen }, (_, idx) =>
      channelFactory({ _id: `channel-${idx}` }, { useConstValues: true }),
    );
  }

  const channels: IChannels = {};
  tmpChannels.forEach(channel => {
    channels[channel._id] = channel;
  });

  let activeChannelInfo =
    Math.random() < 0.5 ? undefined : activeChannelInfoFactory();
  if (options?.useConstValues) activeChannelInfo = undefined;

  return options?.useInitialState
    ? initialState
    : ({
        users,
        channels,
        activeChannelInfo,
        ...(override || {}),
      } as TChatState);
};

export type { IOptions };
