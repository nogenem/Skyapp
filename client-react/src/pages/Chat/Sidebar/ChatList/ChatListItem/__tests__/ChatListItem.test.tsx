import React from 'react';

import { fireEvent } from '@testing-library/dom';
import { render } from '@testing-library/react';

import { IChannel } from '~/redux/chat/types';
import FACTORIES from '~/utils/factories';
import { getRenderWithRedux } from '~/utils/testUtils';

import { ChatListItem, UnconnectedChatListItem } from '../index';

const renderWithRedux = getRenderWithRedux();

describe('Connected ChatListItem', () => {
  it('renders correctly when channel is not a group', () => {
    const channel: IChannel = FACTORIES.models.channel(
      {},
      { useConstValues: true, membersLen: 2 },
    );
    const initialState = {
      chat: FACTORIES.states.chat({
        channels: { [channel._id]: channel },
      }),
    };

    const { container } = renderWithRedux(
      <ChatListItem channelId={channel._id} selected={false} />,
      initialState,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly when channel is a group', () => {
    const channel: IChannel = FACTORIES.models.channel(
      {},
      { useConstValues: true, membersLen: 3 },
    );
    const initialState = {
      chat: FACTORIES.states.chat({
        channels: { [channel._id]: channel },
      }),
    };

    const { container } = renderWithRedux(
      <ChatListItem channelId={channel._id} selected={false} />,
      initialState,
    );

    expect(container).toMatchSnapshot();
  });
});

describe('Unconnected ChatListItem', () => {
  it('calls `emitSetActiveChannel` when clicking on item', () => {
    const channel: IChannel = FACTORIES.models.channel(
      {},
      { useConstValues: true },
    );
    const otherUser = FACTORIES.models.otherUser({}, { useConstValues: true });
    const emitSetActiveChannel = jest.fn();
    const sendLeaveGroupChannel = jest.fn();

    const { getByText } = render(
      <UnconnectedChatListItem
        channelId={channel._id}
        selected={false}
        otherUser={otherUser}
        channel={channel}
        emitSetActiveChannel={emitSetActiveChannel}
        sendLeaveGroupChannel={sendLeaveGroupChannel}
      />,
    );

    fireEvent.click(getByText(channel.name));

    expect(emitSetActiveChannel).toHaveBeenCalled();
    expect(emitSetActiveChannel).toHaveBeenCalledWith(channel._id);
  });
});
