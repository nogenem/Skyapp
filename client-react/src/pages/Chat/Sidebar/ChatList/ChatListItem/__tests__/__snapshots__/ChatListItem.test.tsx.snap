// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected ChatListItem renders correctly when channel is a group 1`] = `
<div>
  <li
    class="MuiListItem-container"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
      role="button"
      tabindex="0"
    >
      <div
        class="makeStyles-avatarContainer-15"
      >
        <div
          class="makeStyles-container-20"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-21"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.99 2c-5.52 0-10 4.48-10 10s4.48 10 10 10 10-4.48 10-10-4.48-10-10-10zm3.61 6.34c1.07 0 1.93.86 1.93 1.93 0 1.07-.86 1.93-1.93 1.93-1.07 0-1.93-.86-1.93-1.93-.01-1.07.86-1.93 1.93-1.93zm-6-1.58c1.3 0 2.36 1.06 2.36 2.36 0 1.3-1.06 2.36-2.36 2.36s-2.36-1.06-2.36-2.36c0-1.31 1.05-2.36 2.36-2.36zm0 9.13v3.75c-2.4-.75-4.3-2.6-5.14-4.96 1.05-1.12 3.67-1.69 5.14-1.69.53 0 1.2.08 1.9.22-1.64.87-1.9 2.02-1.9 2.68zM11.99 20c-.27 0-.53-.01-.79-.04v-4.07c0-1.42 2.94-2.13 4.4-2.13 1.07 0 2.92.39 3.84 1.15-1.17 2.97-4.06 5.09-7.45 5.09z"
              />
            </svg>
          </div>
          <span
            class="makeStyles-dotContainer-22"
          >
            <span
              class="makeStyles-dot-23 makeStyles-dot_active-24 "
              title="Common:Active"
            />
          </span>
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          <span
            class="makeStyles-textContainer-16"
            title="Channel 1"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-noWrap MuiTypography-displayInline"
            >
              Channel 1
            </span>
            <span
              class="MuiTypography-root MuiTypography-caption MuiTypography-colorInherit MuiTypography-displayInline"
            >
              04/12/2021
            </span>
          </span>
        </span>
        <p
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          <span
            class="makeStyles-textContainer-16"
            title="Test message 1"
          >
            <span
              class="MuiTypography-root makeStyles-secondaryText-17 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-displayInline"
            >
              Test message 1
            </span>
            <span
              class="MuiTypography-root MuiTypography-caption MuiTypography-colorInherit MuiTypography-displayInline"
            >
              <span
                class="MuiBadge-root makeStyles-unreadBadge-19"
              >
                <span
                  class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary"
                >
                  2
                </span>
              </span>
            </span>
          </span>
        </p>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiListItemSecondaryAction-root"
    >
      <button
        aria-controls="cli-menu-channel-1"
        aria-haspopup="true"
        aria-label="options"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </li>
</div>
`;

exports[`Connected ChatListItem renders correctly when channel is not a group 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="makeStyles-avatarContainer-1"
    >
      <div
        class="makeStyles-container-6"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
        </div>
        <span
          class="makeStyles-dotContainer-8"
        >
          <span
            class="makeStyles-dot-9 makeStyles-dot_active-10 "
            title="Common:Active"
          />
        </span>
      </div>
    </div>
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        <span
          class="makeStyles-textContainer-2"
          title="Channel 1"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-noWrap MuiTypography-displayInline"
          >
            Channel 1
          </span>
          <span
            class="MuiTypography-root MuiTypography-caption MuiTypography-colorInherit MuiTypography-displayInline"
          >
            04/12/2021
          </span>
        </span>
      </span>
      <p
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        <span
          class="makeStyles-textContainer-2"
          title="Test message 1"
        >
          <span
            class="MuiTypography-root makeStyles-secondaryText-3 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-displayInline"
          >
            Test message 1
          </span>
          <span
            class="MuiTypography-root MuiTypography-caption MuiTypography-colorInherit MuiTypography-displayInline"
          >
            <span
              class="MuiBadge-root makeStyles-unreadBadge-5"
            >
              <span
                class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary"
              >
                2
              </span>
            </span>
          </span>
        </span>
      </p>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;
